### DISCUSSION TITLE
How to publish UDP streams to Ant Media Server?

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5287] [FAQs]
I need to configure Ant Media Server to pull a UDP stream but unfortunately couldn't find a way to do that.
Is it possible to pull a UDP stream using AMS and if yes, then how can I configure it?

**Comment by yashtandon113** [comment]
@Mohit-3196 
Thanks for asking the question.

You can pull the UDP stream on a server as a Stream Source by following below steps.

1. Send the stream to the UDP address using encoder or FFMPEG. In this example, I used FFMPEG command.

`ffmpeg -f lavfi -re -i smptebars=duration=60:size=1280x720:rate=30 -f lavfi -re -i sine=frequency=1000:duration=60:sample_rate=44100 -pix_fmt yuv420p -c:v libx264 -b:v 1000k -g 30 -keyint_min 120 -profile:v baseline -preset veryfast -f mpegts "udp://server-IP:5000?pkt_size=1316"`

You can change the port number as per your requirements.

2. Once the stream started pushing on the server's IP address, you can pull it **(udp://127.0.0.1:5000)** as a stream source on Ant Media Server. To learn how to add stream source on AMS, check [here](https://antmedia.io/docs/guides/publish-live-stream/ip-cameras-and-external-resources/#restreaming-external-sources).
